yt_dlp/postprocessor/ffmpeg.py

```diff
@@ -1173,3 +1186,160 @@ def run(self, info):
             'ext': ie_copy['ext'],
         }]
         return files_to_delete, info
+
+
+class FFmpegProgressTracker:
+
+    def __init__(self, info_dict: dict, ffmpeg_args: List, ffmpeg_proc: Popen, hook_progress: Callable, running_status_value: str, action_bytes_key: str):
+        self.info_dict = info_dict
+        self.ffmpeg_args = ffmpeg_args
+        self.ffmpeg_proc = ffmpeg_proc
+        self.hook_progress = hook_progress
+        self.running_status_value = running_status_value
+        self.action_bytes_key = action_bytes_key
```

Comment:
this seems like a gimick. Better to pass a local hook function here which translates the keys to be suitable for the actual `hook_progress`. We can handle more cases in future like that